source/ida_interface.cc

```diff
@@ -152,6 +152,9 @@ IDAInterface<VEC>::~IDAInterface()
 {
   if (ida_mem)
     IDAFree(&ida_mem);
+#ifdef DEAL_II_WITH_MPI
+  MPI_Comm_free(&communicator);
```

Comment:
Mmmm... I guess this is needed because we don't store a const reference to a communicator, but a *copy* of it, right?